### PR TITLE
ci: add a job to sync 3rd party container images

### PR DESCRIPTION
**Pull request by rchincha**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/sync-3rdparty-images.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions: read-all
+
+jobs:
+  sync-golang:
+    name: 'golang'
```
Comment (smoser) [inline .github/workflows/sync-3rdparty-images.yaml:14]: Where does this get used?

File: .github/workflows/sync-3rdparty-images.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Log in to GitHub Docker Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Tag and push to ghcr
+        run: |
```
Comment (smoser) [inline .github/workflows/sync-3rdparty-images.yaml:31]: does 'run' run with 'set -e' by default?

File: .github/workflows/sync-3rdparty-images.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          docker trust inspect ${{ job.name }}:${{ matrix.version }}
+          docker pull ${{ job.name }}:${{ matrix.version }}
+          docker tag ${{ job.name }}:${{ matrix.version }} ghcr.io/${{ github.repository_owner }}/${{ job.name }}:${{ matrix.version }}
+          docker push ghcr.io/${{ github.repository_owner }}/${{ job.name }}:${{ matrix.version }}
```
Comment (smoser) [inline .github/workflows/sync-3rdparty-images.yaml:35]: Can't we just use one of these run sections and multiple name:versions ?

Or just something like this:

```bash
for n in ubuntu:latest alpine:edge centos:latest; do
    dest="${{ github.repository_owner }}/$n"
    docker trust inspect "$n"
    docker pull $n
    docker tag $n "$dest"
    docker push $dest
done
```

**Review by smoser — CHANGES_REQUESTED** [review]
This feels liek too much text to accomplish what it is doing.  Can we just iterate over a loop of images to sync ? (see inline comment).

I don't actually think this will help much to remove dependency on docker.io.  
1. We still have 'docker://' references in several tests
2. we already were syncing these images to a local oci repo in the beginning of the build.  That sync is *still* using docker:// even after #544, so c-i builds still depend on it.

File: .github/workflows/sync-3rdparty-images.yaml (hunk 3 above)
Comment (rchincha) [reply .github/workflows/sync-3rdparty-images.yaml]: Fixed like so ^

File: .github/workflows/sync-3rdparty-images.yaml (hunk 2 above)
Comment (rchincha) [reply .github/workflows/sync-3rdparty-images.yaml:27]: Added it

**Review by raharper — COMMENTED** [review]
LGTM  -- Can you resolve your issues? Or were you wanting @smoser to reply?

**Review by smoser — APPROVED** [review]

**Pull request merged**